apps/frontend/src/libs/components/select/select.tsx

```diff
@@ -0,0 +1,81 @@
+import {
+	type Control,
+	type FieldPath,
+	type FieldValues,
+} from "react-hook-form";
+import ReactSelect, {
+	type GroupBase,
+	type OptionsOrGroups,
+} from "react-select";
+
+import { getValidClassNames } from "~/libs/helpers/helpers.js";
+import { useFormController } from "~/libs/hooks/hooks.js";
+
+import { type Option } from "./select.js";
+import styles from "./styles.module.css";
+
+type Properties<TFieldValues extends FieldValues, TValue> = {
+	control: Control<TFieldValues, null>;
+	isMulti?: boolean;
+	name: FieldPath<TFieldValues>;
+	options:
+		| OptionsOrGroups<Option<TValue>, GroupBase<Option<TValue>>>
+		| undefined;
+	title?: string;
+};
+
+const Select = <TTFieldValues extends FieldValues, TValue>({
+	control,
+	isMulti = false,
+	name,
+	options,
+	title,
+}: Properties<TTFieldValues, TValue>): JSX.Element => {
```

Comment:
In case you really need this first generic (`TTFieldValue`) it's better to rename it to `TFiedValues` (rid of extra T).
But I'd like to see that this type is not passed throuh props, but instead inferred (generated) inside (cause you have here `TValue` and `isMulti`, so I think it must be possible to do

Reply:
P.s. I'm saying about `const Select <TTFiedValues ..., TValue>(` this first generic

Reply:
Oh, I accidentally typed the second T, thanks.